src/IO/LzmaReadBuffer.h

```diff
@@ -0,0 +1,33 @@
+#pragma once
+
+#include <IO/BufferWithOwnMemory.h>
+#include <IO/CompressionMethod.h>
+#include <IO/ReadBuffer.h>
+
+#include <lzma.h>
+
+namespace DB
+{
+namespace ErrorCodes
+{
+}
+
+class LzmaReadBuffer : public BufferWithOwnMemory<ReadBuffer>
```

Comment:
There is inconsistency between the names:

ZlibInflatingReadBuffer
ZlibDeflatingReadBuffer
LZMAReadBuffer
LZMAWriteBuffer

The inflating and deflating clarification makes the names more clear.
Because it's actually possible to make a WriteBuffer that will decompress data before writing instead of compressing.